core/src/auth_tokens.rs

```diff
@@ -330,4 +439,46 @@ mod tests {
         serde_yaml::from_str::<AuthenticationToken>("{type: \"Bearer\", token: \"AAAA==AAA\"}")
             .unwrap_err();
     }
+
+    #[rstest::rstest]
+    #[case::bearer(r#"{ type: "Bearer", hash: "MJOoBO_ysLEuG_lv2C37eEOf1Ngetsr-Ers0ZYj4vdQ" }"#)]
+    #[case::dap_auth(r#"{ type: "DapAuth", hash: "MJOoBO_ysLEuG_lv2C37eEOf1Ngetsr-Ers0ZYj4vdQ" }"#)]
+    #[test]
+    fn serde_aggregator_token_hash_valid(#[case] yaml: &str) {
+        serde_yaml::from_str::<AuthenticationTokenHash>(yaml).unwrap();
+    }
+
+    #[rstest::rstest]
+    #[case::bearer_token_invalid_encoding(r#"{ type: "Bearer", hash: "+" }"#)]
+    #[case::bearer_token_wrong_length(
+        r#"{ type: "Bearer", hash: "MJOoBO_ysLEuG_lv2C37eEOf1Ngetsr-Ers0ZYj4" }"#
```

Comment:
This case will fail base64url decoding, but I think the intent was to check the conversion from a byte slice to a byte array. I recommending adding another test case with an encoded hash that's one character longer or shorter.

Reply:
`basenc(1)` seems to think it's 30 bytes of valid Base64URL:
```
> echo "MJOoBO_ysLEuG_lv2C37eEOf1Ngetsr-Ers0ZYj4" | basenc -d --base64url | xxd
00000000: 3093 a804 eff2 b0b1 2e1b f96f d82d fb78  0..........o.-.x
00000010: 439f d4d8 1eb6 cafe 12bb 3465 88f8       C.........4e..
> echo "MJOoBO_ysLEuG_lv2C37eEOf1Ngetsr-Ers0ZYj4" | basenc -d --base64url | wc -c
30
```

Reply:
Whoops, I forgot to convert from base64url to base64 when I checked, and Python's base64.b64decode apparently skipped over the underscores and hyphens. The encoded value is in fact 40 bytes long, which is correct.